Source/scrollrt.cpp

```diff
@@ -225,6 +256,68 @@ static void scrollrt_draw_cursor_item()
 	} else {
 		CelClippedDrawSafe(mx + SCREEN_X, my + cursH + SCREEN_Y - 1, pCursCels, pcurs, cursW);
 	}
+
+	// copy cursor from back buffer to a separate bitmap
+	if (hwcursor) {
+		src = &gpBuffer[SCREENXY(sgdwCursX, sgdwCursY)];
+		dst = sgCursorBitmap;
+		for (i = sgdwCursHgt; i != 0; i--, src += BUFFER_WIDTH, dst += sgdwCursWdt) {
+			memcpy(dst, src, sgdwCursWdt);
+		}
+	}
+
+	// remove cursor from the back buffer
+	if (hwcursor) {
+		src = sgSaveBack;
+		dst = &gpBuffer[SCREENXY(sgdwCursX, sgdwCursY)];
+		for (i = sgdwCursHgt; i != 0; i--, src += sgdwCursWdt, dst += BUFFER_WIDTH) {
+			memcpy(dst, src, sgdwCursWdt);
+		}
+	}
+
+	// update system cursor if the bitmap has changed
+	if (hwcursor) {
+		if (sgCursorCel != pcurs || sgCursorCol != col ||
+			sgCursorPaletteVersion != pal_surface_palette_version || sgCursorWindowsResVer != windowResVer) { 
+			// find the actual transparency mask value, fallback to 255
+			if (pass == 1) {
+				char pal[256];

```

Comment:
Why `256`? Is there a const we could use somewhere? Alternatively, could we declare one if one doesn't exist?

Reply:
8 bit graphic have palettes with 256 colors

Reply:
8 bit only has a max of 256 colors, you'd have to convert textures to 2-4 bytes for more colors.
Also I thought index 0 is black and 255 is white. For transparency they used rle but for the UI a specific shade of green